src/main/webapp/search.html

```diff
@@ -4,11 +4,15 @@
     <meta charset="UTF-8">
     <title>Search Page</title>
     <link rel="stylesheet" href="style.css">
-    <script src="script.js"></script>
+    <script src="loginScript.js"></script>
+    <script src="https://apis.google.com/js/platform.js" async defer></script>
+    <meta name="google-signin-client_id" content="12523748853-6h3fnh1kppht9s895ist2b5222ifh2u7.apps.googleusercontent.com">
+
   </head>
-  <body onload="loadBody()">
+  <body>
+    <div class="g-signin2" id="login-button" data-onsuccess="onSignIn"></div>
+    <div class="hidden" id="class-info" >
```

Comment:
FYI it's trivially easy to unhide this in production if the user knows anything about html. So you'll need to make sure you guard the search query on the server side against not-logged-in requests. 

Granted that's something you should be doing anyways for all server endpoints.